clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1399,62 +1399,73 @@ void handleFailure(Throwable throwable) {
 
     @Override
     public DescribeConfigsResult describeConfigs(Collection<ConfigResource> configResources, final DescribeConfigsOptions options) {
-        final Map<ConfigResource, KafkaFutureImpl<Config>> singleRequestFutures = new HashMap<>();
-        final Collection<Resource> singleRequestResources = new ArrayList<>(configResources.size());
-
+        final Map<ConfigResource, KafkaFutureImpl<Config>> allFutures = new HashMap<>(configResources.size());
+        final Map<ConfigResource, KafkaFutureImpl<Config>> unifiedRequestFutures = new HashMap<>();
         final Map<ConfigResource, KafkaFutureImpl<Config>> brokerFutures = new HashMap<>(configResources.size());
+        final Collection<Resource> unifiedRequestResources = new ArrayList<>(configResources.size());
         final Collection<Resource> brokerResources = new ArrayList<>();
 
         for (ConfigResource resource : configResources) {
-            if (resource.type() != ConfigResource.Type.BROKER) {
-                singleRequestFutures.put(resource, new KafkaFutureImpl<Config>());
-                singleRequestResources.add(configResourceToResource(resource));
-            } else {
-                brokerFutures.put(resource, new KafkaFutureImpl<Config>());
+            if (resource.type() == ConfigResource.Type.BROKER) {
+                KafkaFutureImpl<Config> future = new KafkaFutureImpl<>();
+                brokerFutures.put(resource, future);
+                allFutures.put(resource, future);
                 brokerResources.add(configResourceToResource(resource));
+            } else {
+                KafkaFutureImpl<Config> future = new KafkaFutureImpl<>();
+                unifiedRequestFutures.put(resource, future);
+                allFutures.put(resource, future);
+                unifiedRequestResources.add(configResourceToResource(resource));
             }
         }
 
         final long now = time.milliseconds();
-        runnable.call(new Call("describeConfigs", calcDeadlineMs(now, options.timeoutMs()),
+        if (!unifiedRequestResources.isEmpty()) {
```

Comment:
Do I understand correctly that the only behaviour difference apart from `null` checks in case the server response is malformed and the `call` name is this check?

Reply:
This patch also fixes a bug where if a request for a broker configuration failed, we would fail the futures associated with the non-broker request instead (and never fail the broker future, which would be left uncompleted forever)

Reply:
```
        for (Map.Entry<ConfigResource, KafkaFutureImpl<Config>> entry : brokerFutures.entrySet()) {
            final KafkaFutureImpl<Config> brokerFuture = entry.getValue();
        ...
                @Override
                void handleFailure(Throwable throwable) {
                    completeAllExceptionally(singleRequestFutures.values(), throwable);
                }
            }, now);
```

Reply:
Great catch! Do we have a test for this bug? We should include this fix in 0.11.0.1.

Reply:
I added a unit test of a bug where a request for a BROKER config returns an error.